### PR TITLE
Add Google OAuth

### LINKED ISSUE #215: Sign and Login via Google
https://github.com/user-attachments/assets/242a0047-4ce6-4a57-83c7-40ba1a9ea068
IG backend is not setup and connected properly along with proper auth.

### PR DESCRIPTION
**Pull request by chennisden**
Closing #288 in favor of this PR (solves merge conflicts)

**What does this PR do?**

Adds google auth

**What issues does this PR fix or reference?**

Resolves #215

### CODE REVIEW

File: store/app/routers/email_signup.py [hunk 1]
```diff
@@ -28,9 +28,9 @@ class DeleteTokenResponse(BaseModel):
     message: str
 
 
-@email_router.post("/signup/create/", response_model=EmailSignUpResponse)
+# POST: Create Signup Token
+@email_signup_router.post("/create/", response_model=EmailSignUpResponse)
```
Comment (codekansas) [inline store/app/routers/email_signup.py:32]: sad :( i think this is a rebase issue

Comment (chennisden) [reply store/app/routers/email_signup.py:32]: yeah, `git merge --squash -Xours` did not automagically resolve everything. dug through the tests and got it working properly now :+1:

File: frontend/src/components/footer/Footer.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ const Footer = () => {
   const showFooter =
     pathname?.startsWith("/browse") === false &&
     pathname?.startsWith("/login") === false &&
-    pathname?.startsWith("/signup") === false;
+    pathname?.startsWith("/register") === false;
```
Comment (Winston-Hsiao) [inline frontend/src/components/footer/Footer.tsx:21]: make sure this is signup

File: frontend/src/components/auth/SignupWithEmail.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ const SignupWithEmail = () => {
   });
 
   const onSubmit = async ({ email }: EmailSignupType) => {
-    const { data, error } = await auth.client.POST("/email/signup/create/", {
+    const { data, error } = await auth.client.POST("/email/create/", {
```
Comment (Winston-Hsiao) [inline frontend/src/components/auth/SignupWithEmail.tsx:29]: keep previous change (current main changes)

File: frontend/src/components/auth/SignupForm.tsx [hunk 4]
```diff
@@ -45,7 +45,7 @@ const SignupForm: React.FC<SignupFormProps> = ({ signupTokenId }) => {
     }
 
     try {
-      const { error } = await auth.client.POST("/users/signup", {
+      const { error } = await auth.client.POST("/users/register", {
```
Comment (Winston-Hsiao) [inline frontend/src/components/auth/SignupForm.tsx:48]: signup

File: frontend/src/gen/api.ts
Comment (Winston-Hsiao) [inline frontend/src/gen/api.ts:1]: make sure to run openapi once you adjust the router naming conventions, i think you can just discard any of your current changes, or pull most recent master, and `git merge master` while on your branch and keep those changes. Let me know if you want me to handle the merge conflicts/rebase

**Review by Winston-Hsiao — CHANGES_REQUESTED** [review]
let me know if you want me to resolve merge conflicts/rebase

**Review by Winston-Hsiao — APPROVED** [review]

**Pull request merged**